src/main/java/org/spongepowered/api/command/conversation/Conversation.java

```diff
@@ -0,0 +1,268 @@
+     *
+     * @return The current question
+     */
+    Optional<Question> getCurrentQuestion();
+
+    /**
+     * Sets the new question, sending them the header, as well as the prompt
+     * of the question.
+     *
+     * @param question The question to set to
+     */
+    void setQuestion(@Nullable Question question);
+
+    /**
+     * Ends the conversation, calling all of the ending handlers, removing the
+     * {@link Conversant}s, and calls the relevant events.
+     *
+     * <p>A question handler should not call this to end a {@link Conversation}
+     * if it is finished normally, but rather return a
+     * {@link QuestionResultType#END}.</p>
+     *
+     * @param endType The type of end this should be
+     * @param cause The cause of this end
+     * @return Whether or not the conversation truly ended
+     */
+    boolean end(ConversationEndType endType, Cause cause);
+
+    /**
+     * Adds the specified {@link Conversant} to the conversation,
+     * with the generic delete all {@link ExternalChatHandler}.
+     *
+     * <p>This sends them the prompt of the current active question.</p>
+     *
+     * @param conversant The conversant to add to the conversation
+     */
+    default void addConversant(Conversant conversant) {
+        addConversant(conversant, ExternalChatHandlers.deleteAll());
+    }
+
+    /**
+     * Adds a {@link Conversant} to the conversation.
+     *
+     * <p>This sends them the prompt of the current active question.</p>
+     *
+     * @param conversant The conversant to add to the conversation
+     * @param externalChatHandler The chat handler to give the conversant
+     */
+    void addConversant(Conversant conversant, ExternalChatHandler externalChatHandler);
+
+    /**
+     * Removes the specified {@link Conversant} from the conversation.
+     *
+     * <p>This is meant for unique use cases, as only one conversant can
+     * answer each question. Once one answers, both will go to the next
+     * question or step.</p>
+     *
+     * <p>If you remove all {@link Conversant}s remember to end the
+     * conversation with {@link #end(ConversationEndType, Cause)}.</p>
+     *
+     * @param conversant The conversant to remove from the conversation
+     */
+    void removeConversant(Conversant conversant);
+
+    /**
+     * Removes every {@link Conversant} from the conversation.
+     *
+     * <p>This does not end the conversation, so keep in mind
+     * you should use {@link #end(ConversationEndType, Cause)} or else this
+     * conversation will just continue without any conversants.</p>
+     */
+    void removeAllConversants();
+
+    /**
+     * If the specified {@link Conversant} is in the conversation it changes
+     * their {@link ExternalChatHandler} to the one you specify. If they are
+     * not in the conversation, instead use {@link #addConversant(Conversant)}.
+     *
+     * @param conversant The conversant which you want to modify
```

Comment:
Good question. Right now nothing, but do you think it's process should be called? Maybe I can add a boolean to the method to determine that. What's your opinion?